### ISSUE TITLE
[CHECKS] CoreOS Clair as a checks plugin

### ISSUE THREAD

**Issue by vsoch** [apptainer/singularity#818]
suggested by @remyd1 we should look into how feasible it would be to have [CoreOS clair](https://github.com/coreos/clair) as an optional plugin check. Given the dependencies, it would and should not be required, but for  a user that wants to use it, maybe? Something like:

```
singularity check --tag clair container.img
ERROR: You do not have clair installed {do x}

singularity check --tag clair container.img
...runs clair :)
```

**Comment by remyd1** [comment]
Hi @vsoch

Indeed it would be great.
However, IMO, I would not use `clair` as a _tag_ for the check because it can be confusing with image labels. It is rather an option than a tag... No ?

**Comment by vsoch** [comment]
The tags for checks are the entrypoint to selecting one or more specific checks. In that there are many things like clair, and the possibility of wanting to combine, I don't think we should add each of those many to be proper arguments, but rather tags (as is implemented now). For example, now if you wanted to run checks flagged for docker, you would do:

```
singularity check --tag docker ubuntu.img
```

**Comment by remyd1** [comment]
Perfect then ! 😉

**Comment by vsoch** [comment]
hey @remyd1 - I've gotten most of this working, and now I have a clair (server I think?) running from a Singularity image. I'm looking now at the docs, and it looks like Clair is made specifically to run over [Docker layers.](https://github.com/coreos/clair/blob/master/Documentation/api_v1.md#post-layers). I'm wondering what you had in mind for integration with Singularity? Do you want this to be a check for docker layers that Singularity is importing? A more "integrated" integration would mean changing their core software to analyze a filesystem on demand (not a compressed layer) which I don't have the bandwidth to really do. Let me know your thoughts on this!

**Comment by vsoch** [comment]
It may just be updating too, I'm not sure - I'm staring at messages like this:

```
{"Event":"finished fetching","Level":"info","Location":"updater.go:242","Time":"2017-07-23 17:03:41.152271","updater name":"debian"}

{"Event":"finished fetching","Level":"info","Location":"updater.go:242","Time":"2017-07-23 17:09:19.495469","updater name":"oracle"}
{"Event":"finished fetching","Level":"info","Location":"updater.go:242","Time":"2017-07-23 17:10:44.412872","updater name":"ubuntu"}
```
But if this is an update process that must happen, imho it's waaay too slow to warrant for a check!

**Comment by vsoch** [comment]
here is the image --> https://github.com/vsoch/singularity-clair

**Comment by vsoch** [comment]
hey @remyd1 - I've looked into this and I don't see a logical way to integrate it into bootstrap - the primary use case is with a server, and at best there is a no longer supported command line tool (that likely will not continue to be developed). Unless you have another idea/suggestion, I'd like to close this issue.

**Comment by remyd1** [comment]
Hi @vsoch

Ok. Sorry, I did not had any time to look deeply into it to check what
could have been done for singularity.

Sincerely
Rémy

Le 29 juil. 2017 05:10, "Vanessa Sochat" <notifications@github.com> a
écrit :

hey @remyd1 <https://github.com/remyd1> - I've looked into this and I don't
see a logical way to integrate it into bootstrap - the primary use case is
with a server, and at best there is a no longer supported command line tool
(that likely will not continue to be developed). Unless you have another
idea/suggestion, I'd like to close this issue.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/singularityware/singularity/issues/818#issuecomment-318799878>,
or mute the thread
